pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -85,61 +86,74 @@ func (r *DNSPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	dnsPolicy := previous.DeepCopy()
 	log.V(3).Info("DNSPolicyReconciler Reconcile", "dnsPolicy", dnsPolicy)
 
+	// add finalizer to the dnsPolicy
```

Comment:
This is the beginning of a refactor that I did in the main reconcile which I'd like to folks' thoughts on.